src/Neo/SmartContract/Native/NeoToken.cs

```diff
@@ -294,12 +297,38 @@ private void SetGasPerBlock(ApplicationEngine engine, BigInteger gasPerBlock)
         /// </summary>
         /// <param name="snapshot">The snapshot used to read data.</param>
         /// <returns>The amount of GAS generated.</returns>
+        [Obsolete("Use GetGasPerBlock(DataCache) instead")]
         [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
         public BigInteger GetGasPerBlock(DataCache snapshot)
         {
             return GetSortedGasRecords(snapshot, Ledger.CurrentIndex(snapshot) + 1).First().GasPerBlock;
         }
 
+        [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates | CallFlags.WriteStates)]
+        public BigInteger GetGasPerBlock(ApplicationEngine engine, DataCache snapshot)
```

Comment:
@shargon @roman-khimov please check this, i am try to update the gasperblock without really update it casue it should be a council thing.

Reply:
You can change the arguments without hardfork, because both arguments are native ones

Reply:
But we should not write into storages here... should be done by council

Reply:
Maybe CN can detect when this is made by council and start the new time?

Reply:
Hi shargon, i think my latest commit has addressed your concern:

1. reuse existing GetGasPerBlock
2. avoid write to the store but directly calculate a value
3. ensures the council can still update the gasperblock, and the new value can be used immediately.

Reply:
I intentially have two return currentgasperblolck to make the logic more clear as they are different situations. Code can be optimized if the logic passes review if necessary.